packages/react-instantsearch-dom/src/components/Pagination.js

```diff
@@ -87,6 +88,14 @@ class Pagination extends Component {
     };
   }
 
+  onSelect = page => {
+    const { currentRefinement, onChange, refine } = this.props;
+    refine(page);
+    if (page !== currentRefinement && onChange) {
+      this.props.onChange(page);
```

Comment:
I wonder if it would make more sense to call onChange inside every refine itself to solve this generically

Reply:
Wouldn't this imply that it will also be called when a refinement is made via another widget? This is basically what I'd like to avoid as reported in  #164.

Reply:
No, I meant the onChange prop of each widget could be called when its own refine function is called. This can not be called from another widget

Reply:
OK I see 🙂. Indeed, it would be more generic. Do I have a way to add this on each widget without modifying each of them?

Reply:
It has a quite big impact on the library as a whole, so we will discuss what this means for the whole library as a  team. 

Because there's a generic solution for this (in createConnector's refine method) this is a change we'd better do in a major version. 

One complicated thing about calling onChange ourselves, without making the components "controllable" is that we would also need to accept a value, which will be harder to implement

Reply:
Any chance we could still add it only here for now to see if there is really a need before generalizing?